Bottom: 7a1abfcf3372a3f8e59708892fe5812242e93bbd
Top:    7ccbb79aafb878c3c236aeca4aa7de61ae467b5f
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-26 00:25:17 +0100

Refresh of w-c-split-gathering-routines-f-0

---

diff --git a/src/draw.c b/src/draw.c
index 58b60a1..c01f11b 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -186,35 +186,6 @@ Redraw (void)
 }
 
 static int
-element_callback (const BoxType * b, void *cl)
-{
-  ElementTypePtr element = (ElementTypePtr) b;
-  int *side = cl;
-
-  if (ON_SIDE (element, *side))
-    DrawElementPackage (element);
-  return 1;
-}
-
-static void
-draw_element_name (ElementType *element)
-{
-  if (gui->gui && TEST_FLAG (HIDENAMESFLAG, PCB))
-    return;
-  if (TEST_FLAG (HIDENAMEFLAG, element))
-    return;
-  if (doing_pinout || doing_assy)
-    gui->set_color (Output.fgGC, PCB->ElementColor);
-  else if (TEST_FLAG (SELECTEDFLAG, &ELEMENT_TEXT (PCB, element)))
-    gui->set_color (Output.fgGC, PCB->ElementSelectedColor);
-  else if (FRONT (element))
-    gui->set_color (Output.fgGC, PCB->ElementColor);
-  else
-    gui->set_color (Output.fgGC, PCB->InvisibleObjectsColor);
-  DrawTextLowLevel (&ELEMENT_TEXT (PCB, element), PCB->minSlk);
-}
-
-static int
 name_callback (const BoxType * b, void *cl)
 {
   TextTypePtr text = (TextTypePtr) b;
@@ -973,6 +944,61 @@ text_callback (const BoxType * b, void *cl)
   return 1;
 }
 
+static void
+draw_element_package (ELement)
+{
+  /* set color and draw lines, arcs, text and pins */
+  if (doing_pinout || doing_assy)
+    gui->set_color (Output.fgGC, PCB->ElementColor);
+  else if (TEST_FLAG (SELECTEDFLAG, Element))
+    gui->set_color (Output.fgGC, PCB->ElementSelectedColor);
+  else if (FRONT (Element))
+    gui->set_color (Output.fgGC, PCB->ElementColor);
+  else
+    gui->set_color (Output.fgGC, PCB->InvisibleObjectsColor);
+
+  /* draw lines, arcs, text and pins */
+  ELEMENTLINE_LOOP (Element);
+  {
+    _draw_line (line);
+  }
+  END_LOOP;
+  ARC_LOOP (Element);
+  {
+    _draw_arc (arc);
+  }
+  END_LOOP;
+}
+
+static int
+element_callback (const BoxType * b, void *cl)
+{
+  ElementTypePtr element = (ElementTypePtr) b;
+  int *side = cl;
+
+  if (ON_SIDE (element, *side))
+    draw_element_package (element);
+  return 1;
+}
+
+static void
+draw_element_name (ElementType *element)
+{
+  if (gui->gui && TEST_FLAG (HIDENAMESFLAG, PCB))
+    return;
+  if (TEST_FLAG (HIDENAMEFLAG, element))
+    return;
+  if (doing_pinout || doing_assy)
+    gui->set_color (Output.fgGC, PCB->ElementColor);
+  else if (TEST_FLAG (SELECTEDFLAG, &ELEMENT_TEXT (PCB, element)))
+    gui->set_color (Output.fgGC, PCB->ElementSelectedColor);
+  else if (FRONT (element))
+    gui->set_color (Output.fgGC, PCB->ElementColor);
+  else
+    gui->set_color (Output.fgGC, PCB->InvisibleObjectsColor);
+  DrawTextLowLevel (&ELEMENT_TEXT (PCB, element), PCB->minSlk);
+}
+
 
 /* ---------------------------------------------------------------------------
  * draws one non-copper layer
@@ -1452,40 +1478,16 @@ DrawElementName (ElementTypePtr Element)
 void
 DrawElementPackage (ElementTypePtr Element)
 {
-  if (Gathering)
-    {
-      ELEMENTLINE_LOOP (Element);
-      {
-        AddPart (line);
-      }
-      END_LOOP;
-      ARC_LOOP (Element);
-      {
-        AddPart (arc);
-      }
-      END_LOOP;
-      return;
-    }
-
-  /* set color and draw lines, arcs, text and pins */
-  if (doing_pinout || doing_assy)
-    gui->set_color (Output.fgGC, PCB->ElementColor);
-  else if (TEST_FLAG (SELECTEDFLAG, Element))
-    gui->set_color (Output.fgGC, PCB->ElementSelectedColor);
-  else if (FRONT (Element))
-    gui->set_color (Output.fgGC, PCB->ElementColor);
-  else
-    gui->set_color (Output.fgGC, PCB->InvisibleObjectsColor);
+  assert (Gathering);
 
-  /* draw lines, arcs, text and pins */
   ELEMENTLINE_LOOP (Element);
   {
-    _draw_line (line);
+    AddPart (line);
   }
   END_LOOP;
   ARC_LOOP (Element);
   {
-    _draw_arc (arc);
+    AddPart (arc);
   }
   END_LOOP;
 }
@@ -1816,8 +1818,8 @@ DrawObject (int type, void *ptr1, void *ptr2)
 static void
 draw_element (ElementTypePtr element)
 {
-  DrawElementPackage (element);
   draw_element_name (element);
+  draw_element_package (element);
   DrawElementPinsAndPads (element);
 }
